src/handlers.rs

```diff
@@ -27,6 +27,26 @@ impl fmt::Display for HandlerError {
     }
 }
 
+#[derive(Debug)]
+pub struct UserError(String);
+
+impl std::error::Error for UserError {}
+
+impl fmt::Display for UserError {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        f.write_str(&self.0)
+    }
+}
+
+macro_rules! inform {
```

Comment:
Could you add a comment on top of this that explains what it does, or rather how/when it should be used?

Reply:
Sure, comment added and renamed to `user_error!`.